django/contrib/staticfiles/management/commands/collectstatic.py

```diff
@@ -218,7 +218,15 @@ def clear_dir(self, path):
                          smart_text(fpath), level=1)
             else:
                 self.log("Deleting '%s'" % smart_text(fpath), level=1)
-                self.storage.delete(fpath)
+                full_path = self.storage.path(fpath)
+                if not os.path.exists(full_path) and os.path.lexists(full_path):
+                    # Delete if the file is a broken symlink (see #25346)
+                    try:
+                        os.unlink(full_path)
+                    except OSError as e:
+                        CommandError(e)
```

Comment:
In what case is this branch reached? Should there be a test for it? I guess you meant `raise CommandError(e)`.